editor/plugins/org.fusesource.ide.project/src/org/fusesource/ide/project/camel/CamelFacetInstallationDelegate.java

```diff
@@ -0,0 +1,245 @@
+				structureEdit.dispose();
+		}
+		return new ComponentResource[]{};
+	}
+	
+	private IFolder findFolder(IContainer[] p, final String name) {
+		final IFolder[] found = new IFolder[1]; 
+		
+		for( int i = 0; i < p.length && found[0] == null; i++ ) {
+			try {
+				p[i].accept(new IResourceVisitor() {
+					public boolean visit(IResource resource) throws CoreException {
+						if( resource instanceof IFolder && resource.getName().equals(name))
+							found[0] = (IFolder)resource;
+						return found[0] == null;
+					}
+				});
+			} catch(CoreException ce) {
+			}
+		}
+		return found[0];
+	}
+	
+	
+	private void createUtilityProjectReorganized(IProject project) throws CoreException{
+		String strContentFolder = model.getStringProperty(ICamelFacetDataModelProperties.CAMEL_CONTENT_FOLDER);
+		project.setPersistentProperty(ICamelFacetDataModelProperties.QNAME_CAMEL_CONTENT_FOLDER, strContentFolder);
+		
+		IFolder camelContent = project.getFolder(strContentFolder);
+		IProgressMonitor monitor = new NullProgressMonitor();
+		
+		IVirtualComponent newComponent = ComponentCore.createComponent(project);
+		final IVirtualFolder jbiRoot = newComponent.getRootFolder();
+
+		// Map the CAMELcontent to root for deploy
+		String resourcesFolder = model.getStringProperty(ICamelFacetDataModelProperties.CAMEL_CONTENT_FOLDER);
+		jbiRoot.createLink(new Path("/" + resourcesFolder), 0, null); //$NON-NLS-1$
+
+		IFolder metainf = project.getFolder("src").getFolder("META-INF"); 
+		if( metainf.exists())
+			metainf.move(camelContent.getFolder("META-INF").getFullPath(), true, new NullProgressMonitor());
+		createUtilityManifest(project, camelContent.getFolder("META-INF"), monitor);
+		
+		String dsl = model.getStringProperty(ICamelFacetDataModelProperties.CAMEL_DSL);
+		if( dsl.equalsIgnoreCase("Blueprint") ) {
+			IFolder osgiInf = camelContent.getFolder(ICamelFacetDataModelProperties.OSGI_INF); 
+			createFolder(osgiInf, monitor);
+			project.refreshLocal(IResource.DEPTH_ZERO, null);
+			createBlueprintDescriptor(osgiInf);
+		} else if (dsl.equalsIgnoreCase("Spring")) {
+			createSpringDescriptor(camelContent.getFolder("META-INF"));
+		} else if (dsl.equalsIgnoreCase("Routes")) {
+			createRoutesDescriptor(camelContent.getFolder("META-INF"));
+		}
+		
+	}
+	
+	private void createUtilityManifest(IProject project, IFolder metainf, IProgressMonitor monitor) throws CoreException {
+		IFile manifestFile = metainf.getFile("MANIFEST.MF");
+		StringBuffer sb = new StringBuffer();
+		
+		sb.append("Manifest-Version: 1.0\n");
+		sb.append("Bundle-ManifestVersion: 2\n");
+		sb.append("Bundle-Name: " + project.getName() + "\n");
+		sb.append("Bundle-SymbolicName: " + project.getName() +"\n"); // TODO this string won't be osgi-valid as a symbolic name possibly
+		sb.append("Bundle-Version: 1.0.0.SNAPSHOT\n");
+		if( manifestFile.exists())
+			manifestFile.setContents(new ByteArrayInputStream(sb.toString().getBytes()), IResource.FORCE, monitor);
```

Comment:
better using getBytes(StandardCharsets.UTF_8) it will avoid issues (especially on Windows) if the project name is using some special characters.